tests/integration/minicluster_v2.go

```diff
@@ -0,0 +1,261 @@
+		etcdConfig.EtcdTLSCert.GetValue(),
+		etcdConfig.EtcdTLSKey.GetValue(),
+		etcdConfig.EtcdTLSCACert.GetValue(),
+		etcdConfig.EtcdTLSMinVersion.GetValue())
+	if err != nil {
+		return nil, err
+	}
+	cluster.EtcdCli = etcdCli
+
+	cluster.MetaWatcher = &EtcdMetaWatcher{
+		rootPath: etcdConfig.RootPath.GetValue(),
+		etcdCli:  cluster.EtcdCli,
+	}
+
+	// setup clients
+	cluster.RootCoordClient, err = grpcrootcoordclient.NewClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+	cluster.DataCoordClient, err = grpcdatacoordclient.NewClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+	cluster.QueryCoordClient, err = grpcquerycoordclient.NewClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	cluster.ProxyClient, err = grpcproxyclient.NewClient(ctx, paramtable.Get().ProxyGrpcClientCfg.GetInternalAddress(), 0)
+	if err != nil {
+		return nil, err
+	}
+	cluster.DataNodeClient, err = grpcdatanodeclient.NewClient(ctx, paramtable.Get().DataNodeGrpcClientCfg.GetAddress(), 0)
+	if err != nil {
+		return nil, err
+	}
+	cluster.QueryNodeClient, err = grpcquerynodeclient.NewClient(ctx, paramtable.Get().QueryNodeGrpcClientCfg.GetAddress(), 0)
+	if err != nil {
+		return nil, err
+	}
+	cluster.IndexNodeClient, err = grpcindexnodeclient.NewClient(ctx, paramtable.Get().IndexNodeGrpcClientCfg.GetAddress(), 0, false)
+	if err != nil {
+		return nil, err
+	}
+
+	// setup servers
+	cluster.factory = dependency.NewDefaultFactory(true)
+	chunkManager, err := cluster.factory.NewPersistentStorageChunkManager(cluster.ctx)
+	if err != nil {
+		return nil, err
+	}
+	cluster.ChunkManager = chunkManager
+
+	cluster.RootCoord, err = grpcrootcoord.NewServer(ctx, cluster.factory)
+	if err != nil {
+		return nil, err
+	}
+	cluster.DataCoord = grpcdatacoord.NewServer(ctx, cluster.factory)
+	if err != nil {
+		return nil, err
+	}
+	cluster.QueryCoord, err = grpcquerycoord.NewServer(ctx, cluster.factory)
+	if err != nil {
+		return nil, err
+	}
+	cluster.Proxy, err = grpcproxy.NewServer(ctx, cluster.factory)
+	if err != nil {
+		return nil, err
+	}
+	cluster.DataNode, err = grpcdatanode.NewServer(ctx, cluster.factory)
+	if err != nil {
+		return nil, err
+	}
+	cluster.QueryNode, err = grpcquerynode.NewServer(ctx, cluster.factory)
+	if err != nil {
+		return nil, err
+	}
+	cluster.IndexNode, err = grpcindexnode.NewServer(ctx, cluster.factory)
+	if err != nil {
+		return nil, err
+	}
+	return cluster, nil
+}
+
+func (cluster *MiniClusterV2) Start() error {
+	log.Info("mini cluster start")
+	err := cluster.RootCoord.Run()
+	if err != nil {
+		return err
+	}
+
+	err = cluster.DataCoord.Run()
+	if err != nil {
+		return err
+	}
+
+	err = cluster.QueryCoord.Run()
+	if err != nil {
+		return err
+	}
+
+	err = cluster.DataNode.Run()
+	if err != nil {
+		return err
+	}
+
+	err = cluster.QueryNode.Run()
+	if err != nil {
+		return err
+	}
+
+	err = cluster.IndexNode.Run()
+	if err != nil {
+		return err
+	}
+
+	err = cluster.Proxy.Run()
+	if err != nil {
+		return err
+	}
+
```

Comment:
Do we need to check and wait for the service to become healthy?

Reply:
Nice to have it

Reply:
fixed it